yt_dlp/extractor/ceskatelevize.py

```diff
@@ -99,11 +101,12 @@ def _real_extract(self, url):
             playlist_description = playlist_description.replace('\xa0', ' ')
 
         if parsed_url.path.startswith('/porady/'):
-            refer_url = update_url_query(unescapeHTML(self._search_regex(
-                (r'<span[^>]*\bdata-url=(["\'])(?P<url>(?:(?!\1).)+)\1',
-                 r'<iframe[^>]+\bsrc=(["\'])(?P<url>(?:https?:)?//(?:www\.)?ceskatelevize\.cz/ivysilani/embed/iFramePlayer\.php.*?)\1'),
-                webpage, 'iframe player url', group='url')), query={'autoStart': 'true'})
-            webpage = self._download_webpage(refer_url, playlist_id)
+            next_data = self._parse_json(unescapeHTML(self._search_regex(r'<script[^>]+id=[\'"]__NEXT_DATA__[\'"][^>]+>([^<]*)</script>', webpage, 'IDEC id', playlist_id)), playlist_id)
+            idec = traverse_obj(next_data, ('props', 'pageProps', 'data', 'show', 'idec')) or traverse_obj(next_data, ('props', 'pageProps', 'data', 'mediaMeta', 'idec'))
+            if not idec:
+                raise ExtractorError('Failed to find IDEC id')
+            webpage = self._download_webpage(self._IFRAME_HASH_URL, playlist_id)
+            webpage = self._download_webpage(self._IFRAME_URL, playlist_id, query={'hash': webpage, 'origin': 'iVysilani', 'autoStart': 'true', 'IDEC': idec})
```

Comment:
```suggestion
            hash = self._download_webpage(self._IFRAME_HASH_URL, playlist_id)
            webpage = self._download_webpage(self._IFRAME_URL, playlist_id, query={'hash': hash, 'origin': 'iVysilani', 'autoStart': 'true', 'IDEC': idec})
```

Also, since `_IFRAME_HASH_URL` and `_IFRAME_URL` are only ever used here, just hard-code it here, not as class vars

Reply:
OK